packages/@lwc/synthetic-shadow/src/shared/faux-elements-from-point.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isNull, isUndefined } from '@lwc/shared';
+import { elementsFromPoint } from '../env/document';
+import { isSyntheticSlotElement } from '../faux-shadow/traverse';
+
+// Walk up the DOM tree, collecting all shadow roots plus the document root
+function getAllRootNodes(node: Node) {
+    const rootNodes = [];
+    let currentRootNode = node.getRootNode();
+    while (!isUndefined(currentRootNode)) {
+        rootNodes.push(currentRootNode);
+        currentRootNode = (currentRootNode as ShadowRoot).host?.getRootNode();
+    }
+    return rootNodes;
+}
+
+export function fauxElementsFromPoint(
+    context: Node,
+    doc: Document,
+    left: number,
+    top: number
+): Element[] {
+    if (process.env.NODE_ENV !== 'production') {
+        /* eslint-disable-next-line no-console */
+        console.warn(
+            'elementsFromPoint() may return unpredictable results in different browsers, and is not recommended. ' +
+                "Only use this API if you know what you're doing (or this is ChromeDriver, in which case " +
+                'cross-browser compatibility is not a concern).'
+        );
```

Comment:
I'm not sure if we really want `console.warn`, since this will actually write to the Node console when running the Karma tests. But I couldn't think of any other way to put this in here.

Reply:
I don't have a strong opinion either way. @jodarove I think you suggested the warning message; what do you think?